bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/XmlSequence.java

```diff
@@ -675,6 +675,14 @@ public boolean equals(Object o, Set<ValuePair> visitedValues) {
             return this.getChildrenList().size() == 1 &&
                     isEqual(this.getChildrenList().get(0), o);
         }
+        if (o instanceof XmlText rhsXMLText) {
```

Comment:
I'm not sure if this is the correct fix. Shouldn't this have been an xml:Text value to start with, and not a sequence?

Reply:
Are you asking about the fix for the issue that is tagged in the PR (#42836)?

Reply:
Basically, I'm thinking we shouldn't have XML sequences with only xml:Text members.

Reply:
Ok I see your point. I'll look into that.

Reply:
Made changes to the `xmlSequence` to concatenate adjacent `xml:Text` elements together and to the `equals` method to equate `xml:Text` and `xmlSequence` values that has a singleton `xml:Text` element. Please let me know what you think.